src/main/resources/db/migration/V0002__purge.sql

```diff
@@ -0,0 +1,209 @@
+                       FROM sdt_owner.bulk_submissions
+                       WHERE service_request_id IN (
+                           SELECT service_request_id
+                           FROM sdt_owner.service_requests
+                           WHERE request_timestamp < t_retention_date)));
+
+  n_iteration   INT;
+  BEGIN
+
+    -- retention period
+	SELECT parameter_value
+    INTO i_retention_period
+    FROM sdt_owner.global_parameters
+    WHERE parameter_name = 'DATA_RETENTION_PERIOD';
+
+    -- job start date
+	SELECT now()
+    INTO t_job_start_date;
+    RAISE NOTICE 'Job start date: %', t_job_start_date;
+
+    -- retention_date calculated using retention period and now
+	SELECT t_job_start_date::DATE - (i_retention_period || ' DAY')::INTERVAL
+    INTO t_retention_date;
+    RAISE NOTICE 'Retention period date: %', t_retention_date;
+
+    -- create job audit
+	INSERT INTO sdt_owner.purge_job_audit
+  	( job_start_date, success)
+  	VALUES (t_job_start_date, 0);
+
+    -- get job audit id
+	SELECT max(purge_job_id)
+    INTO i_purge_job_id
+    FROM sdt_owner.purge_job_audit;
+
+    RAISE NOTICE 'Purge retention period is % days, deleting all data prior to %',
+                          i_retention_period,
+                          t_retention_date;
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_id, created_date,log_message)
+    VALUES (i_purge_job_id, now(), 'Purge retention period is ' || i_retention_period ||
+			' days, deleting all data prior to ' || t_retention_date);
+
+    -- delete error logs
+    RAISE NOTICE 'Commencing deletion of error_logs records';
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_id, created_date,log_message)
+    VALUES (i_purge_job_id, now(), 'Commencing deletion of error_logs records');
+    n_iteration := 0;
+    FOR c4_rec IN c4(i_retention_period) LOOP
+      DELETE FROM sdt_owner.error_logs
+      WHERE error_log_id = c4_rec.error_log_id;
+      RAISE NOTICE 'Deleted error_logs : %', c4_rec.error_log_id;
+      n_iteration := n_iteration + 1;
+      IF MOD(n_iteration,i_CommitInterval)=0 THEN
+        /* COMMIT; */
+      END IF;
+    END LOOP;
+    RAISE NOTICE '% records deleted from error_logs table.', n_iteration;
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_id, created_date,log_message)
+    VALUES (i_purge_job_id, now(), n_iteration || ' records deleted from error_logs table.');
+    -- update count of deletions - error logs
+    UPDATE sdt_owner.purge_job_audit
+  	SET count_of_error_logs=n_iteration
+  	WHERE purge_job_id=i_purge_job_id;
+
+    RAISE NOTICE 'Commencing deletion of individual_request records';
+    n_iteration := 0;
+    FOR c3_rec IN c3(i_retention_period) LOOP
+      DELETE FROM sdt_owner.individual_requests
+      WHERE individual_request_id = c3_rec.individual_request_id;
+      RAISE NOTICE 'Deleted individual_request : %', c3_rec.sdt_request_reference;
```

Comment:
The deletion of each individual request needs to be logged as well - see corresponding section in original procedure.

Reply:
Done